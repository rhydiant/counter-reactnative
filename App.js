import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';

import { Provider } from 'react-redux';

import Counter from './src/components/counter';
import store from './src/redux/store';

class App extends Component {
  render() {
    console.log('render App');
    return (
      <Provider store={store}>
        <View style={styles.container}>
          <Counter />
        </View>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});

export default App;
