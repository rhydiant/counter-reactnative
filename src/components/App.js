import React from 'react';
import { StyleSheet, View } from 'react-native';

import { Provider } from 'react-redux';

import Counter from './Counter';
import store from '../redux/store';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});

export default function () {
  return (
    <Provider store={store}>
      <View style={styles.container}>
        <Counter />
      </View>
    </Provider>
  );
}
