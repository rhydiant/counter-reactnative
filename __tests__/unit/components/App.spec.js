import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../../src/components/App';

describe('app', () => {
  it.skip('should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
