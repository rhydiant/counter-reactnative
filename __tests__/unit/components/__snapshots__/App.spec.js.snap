// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFD",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text>
      count is 0
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          increment
        </Text>
      </View>
    </View>
  </View>
</View>
`;
